### ISSUE TITLE
endgame still passes unnecessarily sometimes!

### ISSUE THREAD

**Issue by domino14** [domino14/macondo#45]
```
macondo> endgame 10
plies 10, deepening true, simpleEval false, pruningDisabled false

   A B C D E F G H I J K L M N O                        Carl    HOPS?  344
   ------------------------------    ->                 Noah  AFLNNNT  362
 1|=     '       =       '     =
 2|  -       "       "       -      Bag + unseen: (0)
 3|    - l     '   '       -
 4|'   A I       '       -     '
 5|    N O -           -
 6|  Z I N   "       "       "
 7|U   L I     ' C '       '
 8|R   I S     Q A T S   '     G
 9|E   T E   V I D E O     J   R
10|A B Y     "       Y     A M I    Turn 27:
11|  R   O O T H E C A E   W I N    Carl played M9 JAW for 32 pts from a rack
12|B R O K E   U M     X U   D '    of ?AHJPSW
13|U   -     E G O '     P O D
14|R A I A   W E T   A E     L
15|L   D I G   R E F T   V E E S

{"level":"info","time":"2020-04-12T13:44:36-04:00","message":"Spread swing estimate found after 1 plies: 41"}
{"level":"info","time":"2020-04-12T13:44:36-04:00","message":"Spread swing estimate found after 2 plies: -11"}
{"level":"info","time":"2020-04-12T13:44:36-04:00","message":"Spread swing estimate found after 3 plies: -18"}
{"level":"info","time":"2020-04-12T13:44:36-04:00","message":"Spread swing estimate found after 4 plies: -18"}
{"level":"info","time":"2020-04-12T13:44:42-04:00","message":"Spread swing estimate found after 5 plies: -18"}
{"level":"info","time":"2020-04-12T13:44:45-04:00","message":"Spread swing estimate found after 6 plies: -18"}
{"level":"info","time":"2020-04-12T13:45:12-04:00","message":"Spread swing estimate found after 7 plies: -19"}
{"level":"info","time":"2020-04-12T13:45:22-04:00","message":"Spread swing estimate found after 8 plies: -19"}
{"level":"info","time":"2020-04-12T13:45:36-04:00","message":"Spread swing estimate found after 9 plies: -19"}
{"level":"info","time":"2020-04-12T13:45:57-04:00","message":"Spread swing estimate found after 10 plies: -19"}
{"level":"info","time":"2020-04-12T13:45:57-04:00","message":"Best spread found: -19"}
Best sequence has a spread difference of -19
Best sequence:
1) 5C ..NFAN
2) 6B ...e
3) J13 L..
4) 4G PO
5) 7C ..T
6) G4 ..H
7) (Pass)
8) (Pass)
9) (Pass)
10) 5C ......S
```

This is from https://www.cross-tables.com/annotated.php?u=31356#28

**Comment by domino14** [comment]
See turn 8) in the sequence above; it is an unnecessary pass. It's not broken but it's annoying.

**Comment by domino14** [comment]
This endgame also seems to prolong the misery unnecessarily. 
After NONFAN, the optimal sequence wins by 1 (for Carl). The sequence above also has Carl winning by 1, but in a much longer way. I thought I had added a parameter that favored shorter endgames over longer endgames (as long as the spread difference is the same) but it doesn't seem to work for this endgame.
```
1) 6B ...e
2) J13 L..
3) I2 PHOS
```

**Comment by domino14** [comment]
See more interesting notes on this endgame:

https://gist.github.com/domino14/ae9c0c6cb144d4c4988cc49cdecc17b1

**Comment by ddugovic** [comment]
We could copy an idea from https://github.com/official-stockfish/Stockfish/pull/925 (two consecutive passes during search -- not in the actual game history -- could evaluate as immediately ending the game).

**Comment by domino14** [comment]
fixed by eaca79d3bdf3660ab901f43e1393a55e57ae6a7f
